src/eth1spec/rlp.py

```diff
@@ -14,270 +14,343 @@
-    len_partial_x = Uint(len(partial_x))
-    if len_partial_x < 56:
-        return bytearray([192 + len_partial_x]) + partial_x
+    joined_encodings = get_joined_encodings(raw_sequence)
+    len_joined_encodings = Uint(len(joined_encodings))
+
+    if len_joined_encodings < 0x38:
+        return bytearray([192 + len_joined_encodings]) + joined_encodings
     else:
-        big_endian = len_partial_x.to_be_bytes()
-        return bytearray([247 + len(big_endian)]) + big_endian + partial_x
+        len_joined_encodings_as_be = len_joined_encodings.to_be_bytes()
+        return (
+            bytearray([247 + len(len_joined_encodings_as_be)])
+            + len_joined_encodings_as_be
+            + joined_encodings
+        )
 
 
-# for a list, recursively call RLP or RLP_inverse
-def s(x: Sequence[RLP]) -> Bytes:
+def get_joined_encodings(raw_sequence: Sequence[RLP]) -> Bytes:
     """
-    Partially encodes `x`, a list of RLP encodable objects, using RLP,
-    excluding the length prefix.
+    Obtain concatenation of rlp encoding for each item in the sequence
+    raw_sequence.
 
     Parameters
     ----------
-    x : `List[RLP]`
-        List to encode with RLP.
+    raw_sequence : `Sequence[RLP]`
+        Sequence to encode with RLP.
 
     Returns
     -------
-    partially_encoded : `eth1spec.eth_types.Bytes`
-        The partially RLP encoded bytes representing `x`.
+    joined_encodings : `eth1spec.eth_types.Bytes`
+        The concatenated RLP encoded bytes for each item in sequence
+        raw_sequence.
     """
-    if verbose:
-        print("s(", x, ")")
-    partially_encoded = bytearray([])
-    for xi in x:
-        partially_encoded += encode(xi)
-    return partially_encoded
+    joined_encodings = bytearray()
+    for item in raw_sequence:
+        joined_encodings += encode(item)
+
+    return joined_encodings
 
 
-# inverses of above
+#
+# RLP Decode
+#
 
 
-def decode(b: Bytes) -> RLP:
+def decode(encoded_data: Bytes) -> RLP:
     """
-    Decodes an integer, byte sequence, or list of RLP encodable objects from
-    the byte sequence `b`, using RLP.
+    Decodes an integer, byte sequence, or list of RLP encodable objects
+    from the byte sequence `encoded_data`, using RLP.
 
     Parameters
     ----------
-    b : `eth1spec.eth_types.Bytes`
+    encoded_data : `eth1spec.eth_types.Bytes`
         A sequence of bytes, in RLP form.
 
     Returns
     -------
-    decoded : `RLP`
-        Object decoded from `b`.
+    decoded_data : `RLP`
+        Object decoded from `encoded_data`.
     """
-    if verbose:
-        print("RLP_inverse(", b.hex(), ")")
-    if len(b) == 0:
-        return bytearray([0x80])
-    if b[0] < 0xC0:  # bytes
-        return R_b_inverse(b)
+    # Raising error as there can never be empty encoded data for any
+    # given raw data (including empty raw data)
+    # RLP Encoding(b'') -> [128]
+    # RLP Encoding([]) -> [192]
+    assert len(encoded_data) > 0
+
+    if encoded_data[0] <= 0xBF:
```

Comment:
```suggestion
    if encoded_data[0] <= 0xbf:
```

Reply:
Is there a reason for going with upper over lower?

Reply:
It looks like in the code is generally upper and in the comments + tests is lower. I'd prefer to use lower everywhere.

Reply:
I specifically changed them to lower case. Black converted them to capital case again. I'll try to configure black so that this doesn't happen.

Reply:
It looks like the Black formatter team took a specific decision to convert lower case hex to capital literals -> https://github.com/psf/black/issues/1692. @lightclient @SamWilsn what do you guys think about this?

Reply:
Ugh that's really annoying. I think I prefer to override black here.

Reply:
It's pretty silly that it allows lower case in byte strings?

Reply:
I give up. You win this battle, Black.